### PR TITLE
Authentication #30

### PR DESCRIPTION
**Pull request by jmnarloch**
Relates to issue #30

### CODE REVIEW

File: auth.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+const (
+	https        = "https"
+	accessToken = "access_token"
+)
+
+type tokenRequest struct {
+	User     string
+	Password string
+	Realm    *string
```
Comment (mikkeloscar) [inline auth.go]: Any reason why this is a pointer?

File: auth.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+func (auth BearerTokenAuthentication) Enrich(req *http.Request) {
+	if auth.token != nil {
+		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", auth.token))
+	}
+}
+
+// OAuth2Authenticator hides all of the details of authenticating against authorization server
+type OAuth2Authenticator struct {
+	url string
+}
```
Comment (mikkeloscar) [inline auth.go]: You could do `type OAuth2Authenticator string` instead, since the struct just has a single field.

File: auth.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	user, err := reader.ReadString('\n')
+	if err != nil {
+		return nil, err
+	}
+	fmt.Print("Enter password: ")
+	password, err := terminal.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println()
```
Comment (mikkeloscar) [inline auth.go]: I think reading username/password should be handled in the cli part of the code and then passed to the auth code.

File: auth.go (hunk 1 above)
Comment (jmnarloch) [inline auth.go]: The realm can be optional in general, so I though a nullable field would make here more sense.

Yet still I am considering whether it's going to be useful at all and whether it should not be removed at this stage and only added if there would be a real need for it.

File: auth.go (hunk 3 above)
Comment (jmnarloch) [inline auth.go]: Ok, I will do the refactoring.

File: auth.go (hunk 2 above)
Comment (jmnarloch) [inline auth.go]: Done.

File: auth.go (hunk 1 above)
Comment (jmnarloch) [inline auth.go]: I've decided to remove the realm at this point.

File: auth.go (hunk 3 above)
Comment (jmnarloch) [inline auth.go]: I've move the logic to the cli.go.

**Comment by jmnarloch** [comment]
@mikkeloscar would you take a second look, I've done the refactoring of the code and minor tweeks.

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**